src/libasr/codegen/asr_to_llvm.cpp

```diff
@@ -4769,6 +4769,73 @@ class ASRToLLVMVisitor : public ASR::BaseVisitor<ASRToLLVMVisitor>
+                llvm::Value *str = CreateLoad(tmp);
+                llvm::AllocaInst *parg = builder->CreateAlloca(character_type, nullptr);
+                builder->CreateStore(str, parg);
+                llvm::Value *str_len = lfortran_str_len(parg);
+                llvm::Type *type = getIntType(4);
+                llvm::Value *ptr = module->getOrInsertGlobal("_character_to_list_itr",
+                    type);
+                ptr = builder->CreateLoad(ptr);
+                module->getNamedGlobal("_character_to_list_itr")->setInitializer(
+                        llvm::ConstantInt::get(context,
+                            llvm::APInt(32, 0)));
+                llvm::Value* i32_one = llvm::ConstantInt::get(context, llvm::APInt(32, 1));
+
+                ASR::List_t* list_type = ASR::down_cast<ASR::List_t>(x.m_type);
+                bool is_array_type_local = false, is_malloc_array_type_local = false;
+                bool is_list_local = false;
+                ASR::dimension_t* m_dims_local = nullptr;
+                int n_dims_local = -1, a_kind_local = -1;
+                llvm::Type* llvm_el_type = get_type_from_ttype_t(list_type->m_type,
+                                            ASR::storage_typeType::Default, is_array_type_local,
+                                            is_malloc_array_type_local, is_list_local, m_dims_local,
+                                            n_dims_local, a_kind_local);
+                std::string type_code = ASRUtils::get_type_code(list_type->m_type);
+                llvm::DataLayout data_layout(module.get());
+                int32_t type_size =  data_layout.getTypeAllocSize(llvm_el_type);
+                llvm::Type* const_list_type = list_api->get_list_type(llvm_el_type, type_code, type_size);
+                llvm::Value* const_list = builder->CreateAlloca(const_list_type, nullptr, "const_list");
+                list_api->list_init(type_code, const_list, *module, 0, 0);
+                uint64_t ptr_loads_copy = ptr_loads;
+                ptr_loads = 1;
+
+                dict_api->set_iterators();
+
+                llvm::BasicBlock *loophead = llvm::BasicBlock::Create(context, "loop.head");
+                llvm::BasicBlock *loopbody = llvm::BasicBlock::Create(context, "loop.body");
+                llvm::BasicBlock *loopend = llvm::BasicBlock::Create(context, "loop.end");
+                this->current_loophead = loophead;
+                this->current_loopend = loopend;
+
+                // head
+                start_new_block(loophead);
+                llvm::Value *cond = builder->CreateICmpSLT(ptr, str_len);
+                builder->CreateCondBr(cond, loopbody, loopend);
+
+                // body
+                start_new_block(loopbody);
+                llvm::Value* item = lfortran_str_copy(str, ptr, ptr);
+                list_api->append(const_list, item, list_type->m_type, *module);
+                ptr = builder->CreateAdd(ptr, i32_one);
```

Comment:
I tried printing the value of `ptr`, and it seems `ptr` is `0` and doesn't get updated here. Any suggestion to fix this?